### ISSUE TITLE
DNS name and IP in input format

### ISSUE THREAD

**Issue by floyd-fuh** [nmap/nmap#2517]
Happy birthday nmap (sorry I'm late)!

**Describe the current behavior**

TL;DR: Support `example.org (93.184.216.34)` as a hostname specification and skip DNS forward lookup

DNS queries (forward lookups) are obviously necessary but often take a lot of time. The use case of running multiple different nmap scans over similar hostlists (-iL) is common (even suggested at some places in the nmap book). However, although we count on DNS server caching we have to rerun all DNS queries each time an nmap scan is started (AFAIK, please let me know if I'm wrong).

Example: First run:

`nmap -iL hostlist.txt -sL -oA out`

With hostlist.txt being:

`example.org`

Output:

```
[...]
Nmap scan report for example.org (93.184.216.34)
[...]
```

However, when I run a similar scan again, it will do the DNS query again, although it would be very easy for me to:

`grep "Nmap scan report for " out.nmap|cut -d " " -f 5- > new_hostlist.txt`

The new_hostlist.txt is then filled with:

`example.org (93.184.216.34)`

But nmap doesn't recognize that format in further scans:

```
$ nmap -sL -iL test.txt
[...]
Failed to resolve "(93.184.216.34)"
```

As input files can also be space delimited it would be necessary to look for the parentheses to check if it's a next hostname or an IP specification so DNS can be skipped.

**Expected behavior**

Nmap would recognize `example.org (93.184.216.34)` as an input host and would skip DNS forward lookup.

**Comment by dmiller-nmap** [comment]
Thanks for this suggestion. We already have an issue filed for speeding up forward DNS lookups (#1451). The functionality you want can be accomplished by creating entries in your system's [`hosts` file](https://en.wikipedia.org/wiki/Hosts_(file)) (usually `/etc/hosts`). Since Nmap currently uses the `getaddrinfo()` function for forward lookups, an entry in the `hosts` file will result in an immediate return of that value without issuing a network DNS query.

**Comment by floyd-fuh** [comment]
It's not practical to add the 5'000 hosts I'm scanning to the /etc/hosts file of the box while other users on the same system are scanning other targets. Also that doesn't work for automated scripts I'm writing... I'm not complaining about speed, I want to avoid DNS lookup altogether.

My suggestion would be that nmap doesn't need to do any getaddrinfo() or similar, as it can just read the IP from the input.

Are there any other reasons why this is not considered? To me it sounds like a simple change but I don't have a clue about the nmap source code so probably there are reasons?
